sunpy/net/fido_factory.py

```diff
@@ -350,16 +351,22 @@ def fetch(self, *query_results, **kwargs):
         Example
         --------
         >>> from sunpy.net.vso.attrs import Time, Instrument
-        >>> unifresp = Fido.search(Time('2012/3/4','2012/3/5'), Instrument('EIT'))  # doctest: +REMOTE_DATA
+        >>> unifresp = Fido.search(Time('2012/3/4','2012/3/5'), Instrument('EIT'))  
+        ...     # doctest: +REMOTE_DATA
         >>> downresp = Fido.fetch(unifresp)  # doctest: +SKIP
         >>> file_paths = downresp.wait()  # doctest: +SKIP
         """
         wait = kwargs.pop("wait", True)
         progress = kwargs.pop("progress", True)
         reslist = []
-        for query_result in query_results:
-            for block in query_result.responses:
-                reslist.append(block.client.fetch(block, **kwargs))
+        if path is None:
+            for query_result in query_results:
+                for block in query_result.responses:
+                    reslist.append(block.client.fetch(block, **kwargs))
+        else:
+            for query_result in query_results:
+                for block in query_result.responses:
+                    reslist.append(block.client.fetch(block, **kwargs, path=path))
```

Comment:
This `if,else` seems overkill since we are copying the same code twice just adding in a path keyword. 

It would be better to handle a path to None and handle that.